### ISSUE TITLE
Feature request - syslog support

### ISSUE THREAD

**Issue by cgb** [Ylianst/MeshCentral#1126]
Hi there,

Thank you for this amazing body of work - it's quickly become a critical part of my small MSP / remote support toolkit.

One thing I'd value is the ability to send global, device group & device event logs via syslog to an external/centralised logging system.  This would help with auditing & integrating with log monitoring tools.

Thanks!

**Comment by Ylianst** [comment]
Interesting idea and not difficult to add, I will take a look at it.

**Comment by Ylianst** [comment]
Just published MeshCentral v0.5.1-i with syslog support. Add the following lines in the settings section of config.json:

```
    "syslog": "meshcentral",
    "syslogjson": "meshcentral-json"
```

You can also put:

```
    "syslog": true,
    "syslogjson": true
```

Which will give you the same results as above. You can have ether or both syslog's enabled. The JSON syslog is probably the best since you get all of the data in each system. If you use the normal syslog, you just get the message string which is not as detailed. Let me know if that works.

![MC2-SysLog](https://user-images.githubusercontent.com/1319013/78710220-ccec0180-78c9-11ea-9618-cc56e672c43f.png)

**Comment by cgb** [comment]
Thank you so much for the quick turn around!  I'm running on CentOS 7, I installed moder-syslog via npm, but experiencing the below error on startup.  I'm not very familiar with NPM/nodejs troubleshooting, is there anything I can provide to help troubleshoot?


```
[mesh@meshcentral ~]$ npm ls modern-syslog
/home/mesh
`-- modern-syslog@1.2.0 

Apr 08 14:37:59 meshcentral meshcentral-json[2361]: MeshCentral v0.5.1-j Server Start
Apr 08 14:37:59 meshcentral node[2351]: Starting meshcentral-json JSON syslog.
Apr 08 14:38:00 meshcentral node[2351]: MeshCentral HTTP redirection server running on port 800.
Apr 08 14:38:00 meshcentral node[2351]: MeshCentral v0.5.1-j, WAN mode, Production mode.
Apr 08 14:38:00 meshcentral meshcentral-json[2371]: MeshCentral v0.5.1-j Server Start
Apr 08 14:38:01 meshcentral node[2351]: ERR: /home/mesh/node_modules/meshcentral/meshcentral.js:1498
Apr 08 14:38:01 meshcentral node[2351]: if (obj.syslogjson) { obj.syslog.log(obj.syslogjson.LOG_INFO, JSON.stringify(event)); }
Apr 08 14:38:01 meshcentral node[2351]: ^
Apr 08 14:38:01 meshcentral node[2351]: TypeError: Cannot read property 'log' of undefined
Apr 08 14:38:01 meshcentral node[2351]: at Object.CreateMeshCentralServer.obj.DispatchEvent (/home/mesh/node_modules/meshcentral/meshcentral.js:1498:41)
Apr 08 14:38:01 meshcentral node[2351]: at InternalFieldObject.ondone (/home/mesh/node_modules/meshcentral/meshcentral.js:1274:21)
Apr 08 14:38:01 meshcentral node[2351]: { Error: Command failed: /usr/bin/node /home/mesh/node_modules/meshcentral --launch 2351


```

**Comment by Ylianst** [comment]
I see the problem. Fixing now.

**Comment by Ylianst** [comment]
Fixed, will be in MeshCentral v0.5.1-k when I next publish it.

**Comment by cgb** [comment]
Tested successfully, thank you!
